onnxruntime/test/common/path_test.cc

```diff
@@ -224,32 +225,32 @@ TEST(PathTest, RelativePathFailure) {
 
 TEST(PathTest, Concat) {
   auto check_concat =
-      [](const std::string& a, const std::string& b, const std::string& expected_a) {
+      [](const optional<const std::string>& a, const std::string& b, const std::string& expected_a, bool expect_throw = false) {
```

Comment:
i think `const optional<string>&` as opposed to `const optional<const string>&` is fine here. the const optional should prevent modification of the contained string.